iterations_runners/iterations_runner.py

```diff
@@ -105,14 +105,14 @@ def usage():
     num_cores = krun_get_num_cores()
 
     # Pre-allocate result lists
-    wallclock_times = [0] * iters
-    cycle_counts = [None] * num_cores
-    aperf_counts = [None] * num_cores
-    mperf_counts = [None] * num_cores
-    for core in xrange(num_cores):
-        cycle_counts[core] = [0] * iters
-        aperf_counts[core] = [0] * iters
-        mperf_counts[core] = [0] * iters
+    def fill(x, n):
+        return [x for _ in range(n)]
+    floats_fill = fill(0.0, iters)
+    wallclock_times = array.array("d", floats_fill)
+    ints_fill = fill(0, iters)
+    cycle_counts = [array.array("L", ints_fill) for _ in range(num_cores)]
+    aperf_counts = [array.array("L", ints_fill) for _ in range(num_cores)]
+    mperf_counts = [array.array("L", ints_fill) for _ in range(num_cores)]
```

Comment:
The `L` type code is an unsigned integer with minimum size 4 bytes. The types of these arrays work by chance of x86_64 because an `unsigned long` happens to be the same size as a `uint64_t` (confirmed with a C program here).

Is there a way to have an array explicitly backed by a `uint64_t` instead? It doesn't look like it :(

Reply:
(our core cycle and a/mperf metrics are `uint64_t`)

Reply:
It bothers me a bit that someone can port Krun to another platform and get bogus results. This will be forgotten I reckon.

Can we either:

 * Fill a regular list with the largest value expressible by a u64.
 * Make the code crash somehow if the assumption is invalid

Neither is pretty, I know.

Reply:
`assert(array.array("L", [1<<64-1])[0] == 1<<64-1)`?

Reply:
Actually, we needn't worry:

```
>>> array.array("L", [1<<64])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: long int too large to convert
```

Reply:
Maybe, but I'd like to check it fails on a 32-bit arch. Do you have one around?

I notice python3 has a `Q` which is `unsigned long long` of min size 8. Sadly, we are Python 2.

Reply:
```
>>> import ctypes
>>> ctypes.sizeof(ctypes.c_ulong)
8
```

What do you think of that instead?

Reply:
The point is: it'll fail as soon as you put a value in it can't handle. So if it has fewer bytes, it'll refuse a value bigger than it can handle. I'm pretty sure we're OK here.

[And no, I don't have anything that old lurking around ;)]

Reply:
Ah, yeah, OK.

Can you add a comment explaining this, and I then I'm happy.

Reply:
OK, pushed.